tests/csi/cstor/volume/provision_test.go

```diff
@@ -0,0 +1,145 @@
+				WithMaxPool(1).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count")
+			cspCount := ops.GetHealthyCSPCount(spcName, 1)
+			Expect(cspCount).To(Equal(1), "while checking cstorpool health count")
+
+		})
+	})
+
+	AfterEach(func() {
+		By("deleting resources created for testing cstor volume provisioning", func() {
+			By("deleting storageclass")
+			err = ops.SCClient.Delete(scName, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting storageclass", scName)
+
+			By("listing spc")
+			spcList, err = ops.SPCClient.List(metav1.ListOptions{})
+			Expect(err).To(BeNil(), "while listing spc clients", spcList)
+
+			By("deleting spc")
+			for _, spc := range spcList.Items {
+				_, err = ops.SPCClient.Delete(spc.Name, &metav1.DeleteOptions{})
+				Expect(err).To(BeNil(), "while deleting the spc's", spc)
+			}
+		})
+	})
+
+	When("cstor pvc with replicacount 1 is created", func() {
+		It("should create cstor volume target pod", func() {
+
+			By("building a pvc")
+			pvcObj, err = pvc.NewBuilder().
+				WithName(pvcName).
+				WithNamespace(nsName).
+				WithStorageClass(scName).
+				WithAccessModes(accessModes).
+				WithCapacity(capacity).Build()
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pvc {%s} in namespace {%s}",
+				pvcName,
+				nsName,
+			)
+
+			By("creating above pvc")
+			_, err = ops.PVCClient.WithNamespace(nsName).Create(pvcObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating pvc {%s} in namespace {%s}",
+				pvcName,
+				nsName,
+			)
+
+			By("verifying target pod count as 1")
+			controllerPodCount := ops.GetPodRunningCountEventually(openebsNamespace, targetLabel, 1)
+			Expect(controllerPodCount).To(Equal(1), "while checking controller pod count")
+
+			By("verifying pvc status as bound")
+			status := ops.IsPVCBound(pvcName)
+			Expect(status).To(Equal(true), "while checking status equal to bound")
+
+			By("deleting above pvc")
+			err := ops.PVCClient.Delete(pvcName, &metav1.DeleteOptions{})
+			Expect(err).To(
+				BeNil(),
+				"while deleting pvc {%s} in namespace {%s}",
+				pvcName,
+				nsName,
+			)
+
+			By("verifying target pod count as 0")
+			controllerPodCount = ops.GetPodRunningCountEventually(openebsNamespace, targetLabel, 0)
+			Expect(controllerPodCount).To(Equal(0), "while checking controller pod count")
+
+			By("verifying deleted pvc")
+			pvc := ops.IsPVCDeleted(pvcName)
+			Expect(pvc).To(Equal(true), "while trying to get deleted pvc")
+
+			By("verifying if cstorvolume is deleted")
+			CstorVolumeLabel := "openebs.io/persistent-volume=" + pvcObj.Spec.VolumeName
+			cvCount := ops.GetCstorVolumeCountEventually(openebsNamespace, CstorVolumeLabel, 0)
+			Expect(cvCount).To(Equal(0), "while checking cstorvolume count")
```

Comment:
In the next PR we should be thinking about making calls to cstorpool & verify if volume got deleted